packages/credential-provider-sso/src/fromSSO.ts

```diff
@@ -36,35 +47,85 @@ export interface FromSSOInit extends SourceProfileInit {
 /**
  * Creates a credential provider that will read from a credential_process specified
  * in ini files.
+ *
+ * The SSO credential provider must support both
+ *
+ * 1. the legacy profile format,
+ * @example
+ * ```
+ * [profile sample-profile]
+ * sso_account_id = 012345678901
+ * sso_region = us-east-1
+ * sso_role_name = SampleRole
+ * sso_start_url = https://www.....com/start
+ * ```
+ *
+ * 2. and the profile format for SSO Token Providers.
+ * @example
+ * ```
+ * [profile sso-profile]
+ * sso_session = dev
+ * sso_account_id = 012345678901
+ * sso_role_name = SampleRole
+ *
+ * [sso-session dev]
+ * sso_region = us-east-1
+ * sso_start_url = https://www.....com/start
+ * ```
  */
 export const fromSSO =
   (init: FromSSOInit & Partial<SsoCredentialsParameters> = {}): CredentialProvider =>
   async () => {
-    const { ssoStartUrl, ssoAccountId, ssoRegion, ssoRoleName, ssoClient } = init;
-    if (!ssoStartUrl && !ssoAccountId && !ssoRegion && !ssoRoleName) {
+    const { ssoStartUrl, ssoAccountId, ssoRegion, ssoRoleName, ssoClient, ssoSession } = init;
+    const profileName = getProfileName(init);
+
+    if (!ssoStartUrl && !ssoAccountId && !ssoRegion && !ssoRoleName && !ssoSession) {
       // Load the SSO config from shared AWS config file.
       const profiles = await parseKnownFiles(init);
-      const profileName = getProfileName(init);
       const profile = profiles[profileName];
 
+      if (profile.sso_session) {
+        const ssoSessions = await loadSsoSessionData(init);
+        const session = ssoSessions[profile.sso_session];
+        const conflictMsg = ` configurations in profile ${profileName} and sso-session ${profile.sso_session}`;
+        if (ssoRegion && ssoRegion !== session.sso_region) {
+          throw new Error(`Conflicting SSO region` + conflictMsg);
+        }
+        if (ssoStartUrl && ssoStartUrl !== session.sso_start_url) {
+          throw new Error(`Conflicting SSO start url` + conflictMsg);
```

Comment:
This error should break credential chain, and conform to CredentialsProviderError
```suggestion
          throw new CredentialsProviderError(`Conflicting SSO start_url` + conflictMsg, false);
```